src/twisted/newsfragments/10149.bugfix

```diff
@@ -0,0 +1 @@
+Removes twisted.protocol.dict (deprecated in twisted 17.9). Fixes deprecation warnings raised by importing deprecated code from twisted or stdlib.
```

Comment:
The `Removes twisted.protocol.dict (deprecated in twisted 17.9)` part should be moved into a separate PR.

And the release notes fragment should be named `TICKE_ID.removal`

and it should be phrased as

`twisted.protocol.dict which was deprecated in 17.9 was now removed.`

-----------

For this PR a `.misc` fragment should be enought.
This is just an internal testing cleanup and end users should not care about this change.

The change is important and the change is recorded in our git history.

But the info from here is for the information that should be useful for people using Twisted, not developing Twisted.

Reply:
Makes sense, I've restored twisted.protocols.dict & associated tests.